### PR TITLE
Improving hit-boxes

### PR DESCRIPTION
**Pull request by stonko1994**
## Description

Our buttons where styled with margins that means that there are dead space between them. This lead to sometimes hard hittable buttons especially on an mobile device. This pull request improve this by working with paddings.

**Please have a look also on the UI and test it (click through everything)**

## Discussion
I hope this doesn't change to much for a minor release?

## TODO
* Issue in live-stream UI

### CODE REVIEW

File: src/scss/skin-modern/components/_button.scss [hunk 1]
```diff
@@ -7,18 +7,31 @@
   background-position: center;
   background-repeat: no-repeat;
   border: 0;
+  // sass-lint:disable no-important
+  box-sizing: content-box !important;
```
Comment (stonko1994) [inline src/scss/skin-modern/components/_button.scss]: The 'global' box-sizing selector 
```scss
  *,
  *::before,
  *::after {
    box-sizing: border-box;
  }
```

in skin.scss has precedence so over this selector so important is needed here. A possible solution would be to move the box-sizing (in skin.scss) outside of the `.#{$prefix}-ui-uicontainer` but this would affect the styling on the entire page where the UI is included. Bootstrap does it too but i don't think it is a perfect solution.

What do you think @protyposis?

File: src/scss/skin-modern/components/_controlbar.scss [hunk 2]
```diff
@@ -8,38 +8,46 @@
...
   }
 
-  .#{$prefix}-controlbar-top {
-    .#{$prefix}-ui-label {
-      font-size: .9em;
-    }
+  .#{$prefix}-ui-container > {
+    .#{$prefix}-container-wrapper > * {
+      &:first-child {
+        padding-left: 2rem;
```
Comment (stonko1994) [inline src/scss/skin-modern/components/_controlbar.scss]: I just found out that this is an issue if the first issue is `hidden`. (Happens in livestreams without `timeShift`).

A solution would be something like this:
```scss
&:not(.#{$prefix}-hidden) {
  padding-left: 2rem;
}
&:not(.#{$prefix}-hidden) ~ :not(.#{$prefix}-hidden) {
  padding-left: $default-button-badding;
}
```

But i feels so hacky that i don't want this. Again do you have any other idea @protyposis. I was thinking about detecting the first visible item in container JS and add another class but this also feels not perfect.

File: src/scss/skin-modern/_skin.scss [hunk 3]
```diff
@@ -53,6 +53,21 @@
   text-align: left;
   user-select: none;
 
+  // align -ui-playbacktimelabel with our play button in the default ui variant
+  // this does not apply for all buttons / elements but it looks weird without
+  // sass-lint:disable force-pseudo-nesting nesting-depth
+  &:not(.#{$prefix}-ui-skin-cast-receiver):not(.#{$prefix}-ui-skin-smallscreen) {
```
Comment (protyposis) [inline src/scss/skin-modern/_skin.scss]: This really doesn't look pretty 😨

File: src/scss/skin-modern/components/_airplaytogglebutton.scss [hunk 4]
```diff
@@ -4,13 +4,17 @@
 .#{$prefix}-ui-airplaytogglebutton {
   @extend %ui-button;
 
-  background-image: svg('assets/skin-modern/images/airplay.svg');
+  &::before {
```
Comment (protyposis) [inline src/scss/skin-modern/components/_airplaytogglebutton.scss]: As discussed offline, it would be nice if the images could remain as is and we insert a dummy element in `::before` for correct sizing.

File: src/scss/skin-modern/components/_button.scss (hunk 1 above)
Comment (protyposis) [reply src/scss/skin-modern/components/_button.scss]: We need to get rid of the `!important`, but let's postpone this for later. Maybe we get an idea in the meantime on how to solve this.

File: src/scss/skin-modern/components/_label.scss [hunk 5]
```diff
@@ -6,6 +6,7 @@
   @include hidden;
 
   cursor: default;
+  padding: .25em;
```
Comment (protyposis) [inline src/scss/skin-modern/components/_label.scss]: Do you think it would be possible to scale the padding via a global skin variable (in `_variables.scss`)? E.g. `$padding: .25em !default;` or `$padding-scale: 1 !default;`.

File: src/scss/skin-modern/components/_listbox.scss [hunk 6]
```diff
@@ -9,10 +9,12 @@
 
     display: block;
     font-size: .8em;
-    height: 100%;
     min-width: 10em;
     padding: .5em;
-    width: 100%;
+
+    &::before {
+      display: none;
```
Comment (protyposis) [inline src/scss/skin-modern/components/_listbox.scss]: Why is this necessary?

File: src/ts/components/playbacktimelabel.ts [hunk 7]
```diff
@@ -28,7 +28,7 @@ export class PlaybackTimeLabel extends Label<PlaybackTimeLabelConfig> {
     super(config);
 
     this.config = this.mergeConfig(config, <PlaybackTimeLabelConfig>{
-      cssClass: 'ui-playbacktimelabel',
+      cssClasses: ['ui-playbacktimelabel'],
```
Comment (protyposis) [inline src/ts/components/playbacktimelabel.ts]: This should be `cssClass`

File: src/ts/components/titlebar.ts [hunk 8]
```diff
@@ -27,8 +28,16 @@ export class TitleBar extends Container<TitleBarConfig> {
       cssClass: 'ui-titlebar',
       hidden: true,
       components: [
-        new MetadataLabel({ content: MetadataLabelContent.Title }),
-        new MetadataLabel({ content: MetadataLabelContent.Description }),
+        new Container({
```
Comment (protyposis) [inline src/ts/components/titlebar.ts]: The `TitleBar` is already a `Container`, why is a `Container` within a `Container` necessary? This looks like a hack.

File: src/ts/components/titlebar.ts [hunk 9]
```diff
@@ -86,4 +95,20 @@ export class TitleBar extends Container<TitleBarConfig> {
     // init
     checkMetadataTextAndUpdateVisibility();
   }
+
+  /**
+   * Collect all Leaves Components within the Container structure (Needed for keepHiddenWithoutMetadata)
+   * TODO: move to container if needed elsewhere
+   */
+  private static getLeavesComponents(parentComponent: Container<ContainerConfig>): Component<ComponentConfig>[] {
```
Comment (protyposis) [inline src/ts/components/titlebar.ts]: `getLeafComponents`

File: src/ts/components/titlebar.ts [hunk 10]
```diff
@@ -86,4 +95,20 @@ export class TitleBar extends Container<TitleBarConfig> {
     // init
     checkMetadataTextAndUpdateVisibility();
   }
+
+  /**
+   * Collect all Leaves Components within the Container structure (Needed for keepHiddenWithoutMetadata)
```
Comment (protyposis) [inline src/ts/components/titlebar.ts]: `Collect all leaves from the component tree...`

File: src/ts/uifactory.ts [hunk 11]
```diff
@@ -238,14 +237,18 @@ export namespace UIFactory {
...
           components: [
-            new MetadataLabel({ content: MetadataLabelContent.Title }),
-            new CastToggleButton(),
-            new VRToggleButton(),
-            new PictureInPictureToggleButton(),
-            new AirPlayToggleButton(),
-            new VolumeToggleButton(),
-            new SettingsToggleButton({ settingsPanel: settingsPanel }),
-            new FullscreenToggleButton(),
+            new Container({
```
Comment (protyposis) [inline src/ts/uifactory.ts]: Again `Container` within `Container` - looks like a hack. There should be a cleaner solution. What is the actual issue why this is needed?

File: src/scss/skin-modern/components/_label.scss (hunk 5 above)
Comment (stonko1994) [reply src/scss/skin-modern/components/_label.scss]: Hm, it would increase complexity to positioning the `seekbar` / `volumeslider`.

File: src/scss/skin-modern/components/_listbox.scss (hunk 6 above)
Comment (stonko1994) [reply src/scss/skin-modern/components/_listbox.scss]: Because the `::before` is blocking content. For buttons without images this would mean that the text is no longer centered.

File: src/ts/components/titlebar.ts (hunk 8 above)
Comment (stonko1994) [reply src/ts/components/titlebar.ts]: This allowed me to apply css similar to the `controlbar` also for the `titlebar`. 
This `css` expect the first level of `.#{$prefix}-ui-container` to represent a 'row' in the UI. Maybe it's no good approach.

File: src/scss/skin-modern/components/_controlbar.scss [hunk 12]
```diff
@@ -8,38 +8,46 @@
...
+      }
 
-    > .#{$prefix}-container-wrapper > * {
-      margin: 0 .5em;
+      &:last-child {
+        padding-right: 2rem;
```
Comment (protyposis) [inline src/scss/skin-modern/components/_controlbar.scss]: Is it a good idea to mix `rem` with `em`?

File: src/scss/skin-modern/components/_button.scss (hunk 1 above)
Comment (protyposis) [reply src/scss/skin-modern/components/_button.scss]: Solved in https://github.com/bitmovin/bitmovin-player-ui/pull/181
